### DISCUSSION TITLE
Regarding get version

### DISCUSSION THREAD

**Discussion by PoojaGaur1708** [DMTF/libspdm#841] [Q&A] [answered]
Hello , Can you pls tell me if requester is supporting only 1.0 version then what will be required spdm version msg received from responder.
In my case , i am receiving (0x5 0x10 0x4 0x0 0x0 0x0 0x1 0x0 0x0).
Can you pls tell me what should be VersionNumberEntry1:<n> bit.

As in below code libspdm_req_get_version.c L155: it is returning false.

153     /* libspdm_negotiate_connection_version will change the spdm_response.
154      * It must be done after append_message_a.*/
155     result = libspdm_negotiate_connection_version(&common_version,
156                                                   spdm_context->local_context.version.spdm_version,
157                                                   spdm_context->local_context.version.spdm_version_count,
158                                                   spdm_response->version_number_entry,
159                                                   spdm_response->version_number_entry_count);
160     if (result == false) {
161         LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO,
162                        "%s - :%d false***\n",__func__,__LINE__));
163 
164         libspdm_reset_message_a(spdm_context);
165         status = LIBSPDM_STATUS_NEGOTIATION_FAIL;
166         goto receive_done;
167     }
168

**Comment by jyao1** [comment, marked as answer]
@PoojaGaur1708 , The data returned is wrong. It means the version number 0x00 0x00.

GET_VERSION is just first command, I guess you may get more issues.

I believe you can check the issue by yourself. My recommendation is:
1) read SPDM spec, to understand the meaning of the returned byte code.
2) run SPDM emu (https://github.com/DMTF/spdm-emu), to see what is the expected byte as reference.

**Reply by PoojaGaur1708** [reply]
Can you pls tell me what should be data returned for version command by responder for only version 1.1 support ?

**Reply by jyao1** [reply]
2 byte version 0x00 0x11

**Comment by steven-bellock** [comment]
> if requester is supporting only 1.0 version then what will be required spdm version msg received from responder.

The versions returned in a `VERSION` response are completely independent of any versions that the Requester supports. If the Responder returns versions that the Requester doesn't support then the Requester shouldn't talk to that Responder.
